py-polars/scripts/check_stacklevels.py

```diff
@@ -0,0 +1,51 @@
+"""
+Check that warnings decorators are set with the correct stacklevel.
+
+Use `find_stacklevel()` instead of setting it manually.
+"""
+import ast
+import subprocess
+import sys
+from ast import NodeVisitor
+
+# Files in which it's OK to set the stacklevel manually.
+# `git ls-files` lists files with forwards-slashes
+# even on Windows, so it's OK to list them like that.
+EXCLUDE = frozenset(["polars/utils/polars_version.py"])
+
+
+class StackLevelChecker(NodeVisitor):
+    def __init__(self, file) -> None:
+        self.file = file
+        self.violations = set()
+
+    def visit_Call(self, node: ast.Call) -> None:
```

Comment:
I think `c` should be lowercase?

Reply:
the class is `ast.Call`, and the format to override visitors is `visit_classname` https://docs.python.org/3/library/ast.html#ast.NodeVisitor.visit

the ast module is a bit odd in this respect, it's part of stdlib but it breaks pep8..

Reply:
Right, maybe add a comment about that? I am sure I will wonder about this again the second time I read this. ^^

Reply:
A nevermind. I see the signature with `ast.Call`. :+1: